server/connector/serenedb_connector.hpp

```diff
@@ -149,20 +152,35 @@ class RocksDBTable final : public axiom::connector::Table {
     sort_order.resize(
       _pk_type->size(),
       axiom::connector::SortOrder{.isAscending = true, .isNullsFirst = false});
-    // TODO(Dronplane) take it from the catalog when available
-    catalog::Column::Id col_id = 0;
-    for (const auto& [name, type] : std::views::zip(
-           collection.RowType()->names(), collection.RowType()->children())) {
-      auto column = std::make_unique<SereneDBColumn>(name, type, col_id++);
-      columns.push_back(column.get());
-      _column_map.emplace(name, column.get());
-      _column_handles.push_back(std::move(column));
+    for (const auto& catalog_column : collection.Columns()) {
+      auto serenedb_column = std::make_unique<SereneDBColumn>(
+        catalog_column.name, catalog_column.type, catalog_column.id);
+      columns.push_back(serenedb_column.get());
+      _column_map.emplace(catalog_column.name, serenedb_column.get());
+      _column_handles.push_back(std::move(serenedb_column));
     }
     for (const auto& name : _pk_type->names()) {
       const auto* column = findColumn(name);
       SDB_ASSERT(column, "RocksDBTable: can't find PK column ", name);
       order_columns.push_back(column);
     }
+
+    if (_pk_type->children().empty()) {
+      const auto generated_pk_name =
+        catalog::Column::GeneratePKName(collection.RowType()->names());
+
+      auto serenedb_column = std::make_unique<SereneDBColumn>(
+        generated_pk_name, velox::BIGINT(), catalog::Column::kGeneratedPKId);
+      columns.push_back(serenedb_column.get());
+
+      _column_map.emplace(generated_pk_name, serenedb_column.get());
+      _column_handles.push_back(std::move(serenedb_column));
+      order_columns.push_back(findColumn(generated_pk_name));
```

Comment:
we could avoid calling `findColumn`.
We already have desired pointer in `serenedb_column`. If we move this push_back one line up (before we move from `serenedb_column`) we can just take a pointer.